Dockerfile

```diff
@@ -1,25 +1,47 @@
-FROM registry.access.redhat.com/ubi8/python-39 as base
+FROM registry.access.redhat.com/ubi8/ubi-minimal AS build
 
 # Add application sources to a directory that the assemble script expects them
 # and set permissions so that the container runs without root access
 USER 0
 
-RUN dnf -y remove httpd-tools httpd-filesystem
+RUN microdnf install -y python39-devel postgresql-devel gcc && \
+    pip3 install virtualenv                                 && \
+    mkdir -p /opt/app-root                                  && \
+    chown 1001:0 /opt/app-root
 
-ADD app.py setup.py requirements.txt /tmp/src/
-ADD src /tmp/src/src
+USER 1001
+
+COPY --chown=1001:0 app.py setup.py requirements.txt /opt/app-root/
+COPY --chown=1001:0 src /opt/app-root/src
+
+WORKDIR /opt/app-root
+
+RUN virtualenv .                        && \
+    bin/pip install --upgrade pip       && \
+    bin/pip install -r requirements.txt && \
+    bin/python setup.py install         && \
+    rm -rf \~
+
+FROM registry.access.redhat.com/ubi8/ubi-minimal as base
+
+USER 0
+
+WORKDIR /opt/app-root
+
+COPY --chown=1001:0 --from=build /opt/app-root /opt/app-root
+
+RUN microdnf install -y python39 libpq && \
+    chown 1001:0 /opt/app-root
 
-RUN /usr/bin/fix-permissions /tmp/src
 USER 1001
 
-# Install the dependencies
-RUN /usr/libexec/s2i/assemble
+ENV PYTHONPATH=/opt/app-root/build/lib PATH="/opt/app-root/bin:$PATH"
```

Comment:
I don't understand why we need to specify `PYTHONPATH` , could you please try and remove it ? I believe it should work without it .